### ISSUE TITLE
Support SystemJS

### ISSUE THREAD

**Issue by horlescup** [jkuri/ng2-datepicker#50]
Hi,

Can you please support also SystemJS? I'm getting this error (EXCEPTION: Error: Uncaught (in promise): Failed to load ng2-datepicker.component.html) because of relative paths.

The problem is here:
templateUrl: './ng2-datepicker.component.html',
styleUrls: ['./ng2-datepicker.css']

I think the project only works with webpack.

**Comment by teacii** [comment]
Yes pls. same issue

**Comment by jkuri** [comment]
I believe the fastest solution from my side would be including styles and template inline. But that's not okay because I am using sass in dev... 
Can you try searching for that case about SystemJS and Angular templateUrl solution?

**Comment by NightImmunity** [comment]
Yes, the fastest and easiest solution is to include the HTML and CSS inline in the component.  I looked at the way ng2-bootstrap writes their components and it looks like they are doing the same thing.  I will do some research on this if I get time, but for now I have forked and updated my repo to reflect these changes: https://github.com/NightImmunity/ng2-datepicker

**Comment by jasonburrows** [comment]
Same issue.  Just adding my vote to support SystemJS.

**Comment by ghetolay** [comment]
This seems related to #66.

**Comment by corner22** [comment]
Same issue. please fix.

**Comment by lmkerr** [comment]
ditto

**Comment by ghetolay** [comment]
Can any of you guys checkout that the new version is compatible with systemJS ?

Readme about this new version can be found [here](https://github.com/jkuri/ng2-datepicker/tree/2.0.0-dev5).
There is also an example using this new version and systemJS on plunker [here](https://plnkr.co/edit/MgvFblLHi4N3tyLVCWYD?p=preview).

**Comment by jkuri** [comment]
great stuff @ghetolay. only one question... why did you use datepicker-ionic name?

**Comment by ghetolay** [comment]
@jkuri well it's a datepicker and it's using ionic classes and icons. So `datepicker-ionic` seems to fit perfectly.
I don't think I'll have time for that right now but we may have a `datepicker-bootstrap`, `datepicker-material`, `datepicker-minimal` etc... 

At first I thought we could make it a configuration using Input or directive but then I realized we cant configure template nor styles. The only way to do that is by creating new component with different selector.
And I may be wrong but I think each framework have it's own philosophy it's not just about theming so each datepicker may also behave differently.

**Comment by baronbaston** [comment]
For some reason the moduleId in the component declaration is the string `'module.id'` instead of the value `module.id`. Meaning the the html file is still looked for in the root.

On a sidenote. Why are we ment to declare the datepicker ourself as part of the module we are using it from instead of it being defined in its own module which is imported in the consumer module, which I guess is the 'correct' angular way.
